lib/portage/_emirrordist/FetchTask.py

```diff
@@ -88,16 +89,20 @@ def _start(self):
 				# Do the unlink in order to ensure that the path is clear,
 				# even if stat raised ENOENT, since a broken symlink can
 				# trigger ENOENT.
-				if self._unlink_file(distfile_path, "distfiles"):
-					if st is not None:
-						logging.debug(("delete '%s' with "
-							"wrong size from distfiles") % (self.distfile,))
-				else:
-					self.config.log_failure("%s\t%s\t%s" %
-						(self.cpv, self.distfile, "unlink failed in distfiles"))
-					self.returncode = os.EX_OK
-					self._async_wait()
-					return
+				for layout in self.config.layouts:
+					unlink_path = os.path.join(
+						self.config.options.distfiles,
+						layout.get_path(self.distfile))
+					if self._unlink_file(unlink_path, "distfiles"):
+						if st is not None:
+							logging.debug(("delete '%s' with "
+								"wrong size from distfiles") % (self.distfile,))
+					else:
+						self.config.log_failure("%s\t%s\t%s" %
+							(self.cpv, self.distfile, "unlink failed in distfiles"))
+						self.returncode = os.EX_OK
+						self._async_wait()
+						return
```

Comment:
It returns before we've looped through all of the layouts here.

Reply:
I don't think it really matters since it won't do anything useful anyway but sure, changed that.